### PR TITLE
Adapt changes for updated hass client

### LINKED ISSUE #88: Connection issues after restarting Home Assistant
**Describe the bug**

Actually idk why this happened but after this error appeared none of the lights
are responding anymore by Hue Entertainment Area. 

**Steps to Reproduce**

I have no Idea?
Maybe some Sort of not closing a connection properly?

**Expected behavior**

**Logs**

```
2021-01-22 00:54:44,322 ERROR asyncio -- Task exception was never retrieved
future: <Task finished name='Task-2612715' coro=<EntertainmentAPI.__async_process_light_packet() done, defined at /app/emulated_hue/entertainment.py:106> exception=TypeError("'NoneType' object is not callable")>
Traceback (most recent call last):
  File "/app/emulated_hue/entertainment.py", line 144, in __async_process_light_packet
    await self.hass.async_call_service("light", "turn_on", svc_data)
  File "/usr/local/lib/python3.8/site-packages/hass_client/__init__.py", line 204, in async_call_service
    return await self.__async_send_ws(msg)
TypeError: 'NoneType' object is not callable
2021-01-22 00:54:44,363 ERROR asyncio -- Task exception was never retrieved
future: <Task finished name='Task-2612717' coro=<EntertainmentAPI.__async_process_light_packet() done, defined at /app/emulated_hue/entertainment.py:106> exception=TypeError("'NoneType' object is not callable")>
Traceback (most recent call last):
  File "/app/emulated_hue/entertainment.py", line 144, in __async_process_light_packet
    await self.hass.async_call_service("light", "turn_on", svc_data)
  File "/usr/local/lib/python3.8/site-packages/hass_client/__init__.py", line 204, in async_call_service
    return await self.__async_send_ws(msg)
TypeError: 'NoneType' object is not callable
2021-01-22 00:54:44,404 ERROR asyncio -- Task exception was never retrieved
future: <Task finished name='Task-2612719' coro=<EntertainmentAPI.__async_process_light_packet() done, defined at /app/emulated_hue/entertainment.py:106> exception=TypeError("'NoneType' object is not callable")>
Traceback (most recent call last):
  File "/app/emulated_hue/entertainment.py", line 144, in __async_process_light_packet
    await self.hass.async_call_service("light", "turn_on", svc_data)
  File "/usr/local/lib/python3.8/site-packages/hass_client/__init__.py", line 204, in async_call_service
    return await self.__async_send_ws(msg)
TypeError: 'NoneType' object is not callable
2021-01-22 00:54:44,445 ERROR asyncio -- Task exception was never retrieved
future: <Task finished name='Task-2612721' coro=<EntertainmentAPI.__async_process_light_packet() done, defined at /app/emulated_hue/entertainment.py:106> exception=TypeError("'NoneType' object is not callable")>
Traceback (most recent call last):
  File "/app/emulated_hue/entertainment.py", line 144, in __async_process_light_packet
    await self.hass.async_call_service("light", "turn_on", svc_data)
  File "/usr/local/lib/python3.8/site-packages/hass_client/__init__.py", line 204, in async_call_service
    return await self.__async_send_ws(msg)
TypeError: 'NoneType' object is not callable
2021-01-22 00:54:44,487 ERROR asyncio -- Task exception was never retrieved
future: <Task finished name='Task-2612723' coro=<EntertainmentAPI.__async_process_light_packet() done, defined at /app/emulated_hue/entertainment.py:106> exception=TypeError("'NoneType' object is not callable")>
Traceback (most recent call last):
  File "/app/emulated_hue/entertainment.py", line 144, in __async_process_light_packet
    await self.hass.async_call_service("light", "turn_on", svc_data)
  File "/usr/local/lib/python3.8/site-packages/hass_client/__init__.py", line 204, in async_call_service
    return await self.__async_send_ws(msg)
TypeError: 'NoneType' object is not callable
2021-01-22 00:54:44,529 ERROR asyncio -- Task exception was never retrieved
future: <Task finished name='Task-2612725' coro=<EntertainmentAPI.__async_process_light_packet() done, defined at /app/emulated_hue/entertainment.py:106> exception=TypeError("'NoneType' object is not callable")>
Traceback (most recent call last):
  File "/app/emulated_hue/entertainment.py", line 144, in __async_process_light_packet
    await self.hass.async_call_service("light", "turn_on", svc_data)
  File "/usr/local/lib/python3.8/site-packages/hass_client/__init__.py", line 204, in async_call_service
    return await self.__async_send_ws(msg)
TypeError: 'NoneType' object is not callable
2021-01-22 00:54:44,550 ERROR asyncio -- Task exception was never retrieved
future: <Task finished name='Task-2612726' coro=<EntertainmentAPI.__async_process_light_packet() done, defined at /app/emulated_hue/entertainment.py:106> exception=TypeError("'NoneType' object is not callable")>
Traceback (most recent call last):
  File "/app/emulated_hue/entertainment.py", line 144, in __async_process_light_packet
    await self.hass.async_call_service("light", "turn_on", svc_data)
  File "/usr/local/lib/python3.8/site-packages/hass_client/__init__.py", line 204, in async_call_service
    return await self.__async_send_ws(msg)
TypeError: 'NoneType' object is not callable
2021-01-22 00:54:44,613 ERROR asyncio -- Task exception was never retrieved
future: <Task finished name='Task-2612729' coro=<EntertainmentAPI.__async_process_light_packet() done, defined at /app/emulated_hue/entertainment.py:106> exception=TypeError("'NoneType' object is not callable")>
Traceback (most recent call last):
  File "/app/emulated_hue/entertainment.py", line 144, in __async_process_light_packet
    await self.hass.async_call_service("light", "turn_on", svc_data)
  File "/usr/local/lib/python3.8/site-packages/hass_client/__init__.py", line 204, in async_call_service
    return await self.__async_send_ws(msg)
TypeError: 'NoneType' object is not callable
2021-01-22 00:54:44,655 ERROR asyncio -- Task exception was never retrieved
future: <Task finished name='Task-2612731' coro=<EntertainmentAPI.__async_process_light_packet() done, defined at /app/emulated_hue/entertainment.py:106> exception=TypeError("'NoneType' object is not callable")>
Traceback (most recent call last):
  File "/app/emulated_hue/entertainment.py", line 144, in __async_process_light_packet
    await self.hass.async_call_service("light", "turn_on", svc_data)
  File "/usr/local/lib/python3.8/site-packages/hass_client/__init__.py", line 204, in async_call_service
    return await self.__async_send_ws(msg)
TypeError: 'NoneType' object is not callable
2021-01-22 00:54:44,676 ERROR asyncio -- Task exception was never retrieved
future: <Task finished name='Task-2612732' coro=<EntertainmentAPI.__async_process_light_packet() done, defined at /app/emulated_hue/entertainment.py:106> exception=TypeError("'NoneType' object is not callable")>
Traceback (most recent call last):
  File "/app/emulated_hue/entertainment.py", line 144, in __async_process_light_packet
    await self.hass.async_call_service("light", "turn_on", svc_data)
  File "/usr/local/lib/python3.8/site-packages/hass_client/__init__.py", line 204, in async_call_service
    return await self.__async_send_ws(msg)
TypeError: 'NoneType' object is not callable
2021-01-22 00:54:44,738 ERROR asyncio -- Task exception was never retrieved
future: <Task finished name='Task-2612735' coro=<EntertainmentAPI.__async_process_light_packet() done, defined at /app/emulated_hue/entertainment.py:106> exception=TypeError("'NoneType' object is not callable")>
Traceback (most recent call last):
  File "/app/emulated_hue/entertainment.py", line 144, in __async_process_light_packet
    await self.hass.async_call_service("light", "turn_on", svc_data)
  File "/usr/local/lib/python3.8/site-packages/hass_client/__init__.py", line 204, in async_call_service
    return await self.__async_send_ws(msg)
TypeError: 'NoneType' object is not callable
2021-01-22 00:54:44,759 ERROR asyncio -- Task exception was never retrieved
future: <Task finished name='Task-2612736' coro=<EntertainmentAPI.__async_process_light_packet() done, defined at /app/emulated_hue/entertainment.py:106> exception=TypeError("'NoneType' object is not callable")>
Traceback (most recent call last):
  File "/app/emulated_hue/entertainment.py", line 144, in __async_process_light_packet
    await self.hass.async_call_service("light", "turn_on", svc_data)
  File "/usr/local/lib/python3.8/site-packages/hass_client/__init__.py", line 204, in async_call_service
    return await self.__async_send_ws(msg)
TypeError: 'NoneType' object is not callable
```

**Hardware Info:**
 - Architecture (Ex: x86, ARM): 
 - Addon version (Ex: stable or dev):  stable
    - Value of "tag_commit_or_branch" if using dev: 
 - OS (Ex: Windows 10, Ubuntu Buster): HyperV Environment on Windows10
 - Hardware (Ex: PC, Raspberry Pi 1/2/3/Zero): PC

**Additional context**

I closed Razer Synapse a few times for testing maybe there got something messed up with the connection and its not closed properly?

### PR DESCRIPTION
**Pull request by marcelveldt**
I've refactored the Hass client library a bit. It should be more robust now and better handle reconnects.
This PR adopts our code to this new Hass Client release.

closes #88

### CODE REVIEW

File: emulated_hue/api.py [hunk 1]
```diff
@@ -670,7 +670,7 @@ async def __async_light_action(self, entity: dict, request_data: dict) -> None:
             )
 
         # execute service
-        await self.hass.async_call_service(const.HASS_DOMAIN_LIGHT, service, data)
+        await self.hass.call_service(const.HASS_DOMAIN_LIGHT, service, data)
```
Comment (alexyao2015) [inline emulated_hue/api.py:673]: Why was this changed to remove async while it's still being awaited?

File: emulated_hue/entertainment.py [hunk 2]
```diff
@@ -56,7 +56,7 @@ async def async_run(self):
         # As a (temporary?) workaround we rely on the OpenSSL executable which is
         # very well supported on all platforms.
         LOGGER.info("Start HUE Entertainment Service on UDP port 2100.")
-        await self.hue.hass.async_set_state(
+        await self.hue.hass.set_state(
```
Comment (alexyao2015) [inline emulated_hue/entertainment.py:59]: Same here?

**Review by alexyao2015 — COMMENTED** [review]
Other than the changes to the notation, as long as it works (which I assume it does but I haven't looked at the changes to the client), it seems great!

File: emulated_hue/api.py (hunk 1 above)
Comment (marcelveldt) [reply emulated_hue/api.py:673]: You asked me a while ago to remove the async_ prefixes so I started here ;-)
These async_ prefixes come from a time asyncio was not mainstream. I'm slowly adjusting my code to not include the prefix anymore...

Comment (alexyao2015) [reply emulated_hue/api.py:673]: Ah right! I thought you had good reason for leaving them in at the time. Typically my IDE will tell me when I'm calling an async method without await, so it was never a huge deal with or without the. I'm fine either way but thought it was a little strange given that you previously mentioned it being important while removing it here.

**Review by alexyao2015 — APPROVED** [review]

File: emulated_hue/api.py (hunk 1 above)
Comment (marcelveldt) [reply emulated_hue/api.py:673]: I do not have a preference anymore. I liked the async_ prefix to make it more clear it's async and to allow a method to have both a sync and a async version. I guess that now we're actually only coding everything async the prefix does no longer make sense. Asyncio even warns you nowadays if you're calling an async method without awaiting it.

Maybe turn things around and if there's a sync method available for some logic, just prefix it with "sync_"

Comment (alexyao2015) [reply emulated_hue/api.py:673]: I don't think adding sync is absolutely necessary ;), but we may look into adding unit tests that could catch issues in the code like this if they arise. I'm going to handle all the dependabot PRs and merge this after. 

Perhaps it's time for a small release after as well.

File: emulated_hue/__init__.py [hunk 3]
```diff
@@ -21,11 +22,13 @@ def __init__(
         hass_token: str,
         http_port: int,
         https_port: int,
-    ):
+    ) -> None:
         """Create an instance of HueEmulator."""
         self._loop = None
         self._config = Config(self, data_path, http_port, https_port)
-        self._hass = HomeAssistant(url=hass_url, token=hass_token)
+        self._hass: Optional[HomeAssistantClient] = None
```
Comment (alexyao2015) [inline emulated_hue/__init__.py:29]: @marcelveldt 
Any reason for doing it like this? Seems like this creates a bug in the latest version as _hass is not defined in __init__ so when `import HueApi` is called, None gets returned causing communication with hass to fail.
